internal/pkg/service/buffer/api/service/service.go

```diff
@@ -186,6 +191,45 @@ func (*service) Import(dependencies.ForPublicRequest, *buffer.ImportPayload, io.
 	return &NotImplementedError{}
 }
 
+func parseRequestBody(contentType string, reader io.ReadCloser) (res *orderedmap.OrderedMap, err error) {
+	// Limit read csv to 1 MB plus 1 B. If the reader fills the limit then the request is bigger than allowed.
+	limitedReader := io.LimitReader(reader, configstore.MaxImportRequestSizeInBytes+1)
+	defer func() {
+		if closeErr := reader.Close(); closeErr != nil && err == nil {
+			err = errors.Errorf("cannot close request body reading: %w", closeErr)
+		}
+	}()
+
+	buf := new(strings.Builder)
+	_, err = io.Copy(buf, limitedReader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that the reader did not read more than the maximum.
+	if buf.Len() > configstore.MaxImportRequestSizeInBytes {
+		return nil, &PayloadTooLargeError{Message: "Payload too large."}
+	}
+
+	var data *orderedmap.OrderedMap
+	if isContentTypeForm(contentType) {
+		data, err = utilsUrl.ParseQuery(buf.String())
+		if err != nil {
+			return nil, &BadRequestError{Message: "Could not parse POST request body into a json."}
+		}
+	} else {
+		err = json.Unmarshal([]byte(buf.String()), &data)
+		if err != nil {
+			return nil, &BadRequestError{Message: "Could not parse form request body into a json."}
+		}
+	}
```

Comment:
Když to není form, tak jde o json. Ve vedlejším PR mám nachystanou úpravu API designu:
![CleanShot 2022-11-22 at 22 38 53@2x](https://user-images.githubusercontent.com/215660/203426553-300aff34-8b87-4707-bbaa-354537c642d4.jpg)